python/cudf/cudf/core/column/column.py

```diff
@@ -570,7 +571,7 @@ def _scatter_by_slice(
         start, stop, step = key.indices(len(self))
         if start >= stop:
             return None
-        num_keys = (stop - start) // step
+        num_keys = ceil((stop - start) / step)
```

Comment:
Should we worry about floating precision here? We can do this entirely as an integral computation.
```suggestion
        num_keys = (stop - start) // step)
        if (stop - start) % step:
            num_keys += 1
```

or even `divmod`, which I _think_ uses only integral math:
```suggestion
        num_keys, remainder = divmod(stop - start, step)
        if remainder:
            num_keys += 1
```

Reply:
How about: `len(range(len(self))[key])`

Reply:
I'm not sure what the danger is w.r.t. floating point precision. Isn't the worst case going to be ((2^^32 - (2^^32-1)) / (2^^32-1)). `ceil` works fine on this value, anyway.

I'm not a fan of `len(range(len(self))[key]`. That has us building python range objects that are as much as 2^^32 in size, major performance crush.

Reply:
@thomcom Indexing a `range` with a `slice` produces another `range`. It won't materialize that as a list of the indices.

Reply:
@wence- has the best solution here afaict. Good call. I should've zoomed out and seen that in context. :)
```suggestion
        num_keys = len(range(len(self))[key])
```

Reply:
Probably it is more obviously written as:
```
len(range(start, stop, step))
```